### DISCUSSION TITLE
breaking special alert window ($csp filter in FF ..)

### DISCUSSION THREAD

**Discussion by peace2000** [uBlockOrigin/uBlock-issues#2081] [General]
Hi, concerning this thread: https://github.com/uBlockOrigin/uBlock-issues/issues/2079 and Gorhill's comment:

`The issue is that the CSP for the page prevent inline scripts from being injected:`

Can I use `$csp` to overcome that limitation? To allow inline scripts?

I have seen this filter syntax around and know somewhat roughly it's purpose but I'm a bit uncertain if that can be used to craft a proper / safe filter for this site: `ylilauta.org`

**Comment by gorhill** [comment]
You can't use `csp` to loosen a page's own CSP restrictions, by design.

**Comment by peace2000** [comment]
Hmm. How then I can use other scriplets to overcome this alert message? (Alert-buster doesn't work without errors for this site even on Chrome anymore.). 

![kuva](https://user-images.githubusercontent.com/17256841/162509014-3e4c1b5b-ed38-441b-b762-56bd3658cfb1.png)

(To trigger this message, FIN list has to be enabled.)

It nags about ads. This code is probably responsible for it:

```
 let alertMessage = message;
                if (title !== '' && title !== null) {
                    alertMessage = title + ': ' + alertMessage;
                }

                if (!this.toastAlertShown) {
                    alert(alertMessage);
                }

                this.toastAlertShown = true;
                setTimeout(() => {
                    this.toastAlertShown = false;
                }, 4_000);

                console.error('Toast could not be displayed properly for some reason, it probably does not fit in the viewport. ' +
                    'Falling back to alert(), browsers should know how to display it properly.');
```

`https://static.ylilauta.org/js/3.12/Library/Toast.js`

That alert is displayed because a toast that nags about ads, is blocked. And that toast appears when ads are blocked. That alert is kind of "back up" nag.

I was earlier able to defuse it using `aopr` on `console.error` but it doesn't work anymore because the location of `console.error` was changed in the code.

Sample link: `https://ylilauta.org/satunnainen/130286730`

**Comment by mapx-** [comment]
You could replace the current filter with:

`||fcdn.lauta.media^$frame,redirect=noopframe,important`

**Reply by peace2000** [reply]
Otherwise it works but there's still a big blank area as a leftover at the top.

**Comment by mapx-** [comment]
in chrome could work (but to check for breakage)

`ylilauta.org##+js(aost, document.querySelector)`

for FF working using `ScriptletDoctor`

**Reply by peace2000** [reply]
Prevents creating new threads, unfortunately.

**Comment by mapx-** [comment]
I guess needing an account to create new threads

test
`ylilauta.org##+js(aost, document.querySelector, , 2)`

and see what you get in the console 

Now I get 
```
stackDepth:2
 Toast https://static.ylilauta.org/js/3.12/Library/Toast.js:12:1
 loadModules/< https://static.ylilauta.org/js/3.12/Lauta.js:282:1
```

When you create a thread it's some other log ?

**Reply by peace2000** [reply]
Firefox:

```
stackDepth:5
 Toast https://static.ylilauta.org/js/3.12/Library/Toast.js:12:1
 loadModules/< https://static.ylilauta.org/js/3.12/Lauta.js:282:1
 promise callback*loadModules https://static.ylilauta.org/js/3.12/Lauta.js:259:1
 Lauta https://static.ylilauta.org/js/3.12/Lauta.js:43:1
 https://static.ylilauta.org/js/3.12/Lauta.js:333:1
```
 
 Chrome:
 
```
 stackDepth:3
 new Toast https://static.ylilauta.org/js/3.12/Library/Toast.js:12:1
 https://static.ylilauta.org/js/3.12/Lauta.js:282:1
 async Lauta.loadModules https://static.ylilauta.org/js/3.12/Lauta.js:259:1
```

You don't need to have an account, the other option is that you have been on the site 24 hrs (based on cookies I think?).

Btw it's possible to change the menus into English in this link: `https://ylilauta.org/user/`

![kuva](https://user-images.githubusercontent.com/17256841/162519533-eca7078f-663c-4de5-8db1-63ce3d331631.png)

I can't create a thread at all with that filter.

**Comment by mapx-** [comment]
test also

`ylilauta.org##+js(aost, document.createElement, , 2)`

**Reply by peace2000** [reply]
Doesn't remove ads.

Chrome on console:

```
Window width was 0, retry 1: 1032
Window width was 0, retry 1: 1008
```

Firefox doesn't report anything special.

**Reply by mapx-** [reply]
You are talking about ads now ? The goal was to remove that alert window.

add & test only `ylilauta.org##+js(aost, document.createElement, , 2)`
(no other filter)

**Reply by peace2000** [reply]
That filter seems to prevent opening images as bigger.

**Reply by peace2000** [reply]
That alert doesn't activate if this

![kuva](https://user-images.githubusercontent.com/17256841/162525570-dd258db1-ee70-4d73-b7d5-f40f6b9bae6a.png)

is hid without the site noticing it. Is there a way to do that, maybe?

**Reply by mapx-** [reply]
test this filter

`ylilauta.org##+js(aost, document.createElement, /loadModule|new/, 2)`

**Reply by peace2000** [reply]
That seems to work, will test it for a few days. Thanks!

**Reply by peace2000** [reply]
Any guesses what breakages it could cause, if any?

**Reply by mapx-** [reply]
If you get some other breakage I'll try to exclude it limiting `aost` only to those toasts (not always possible ..)

**Reply by peace2000** [reply]
Found a breakage now. Cannot open notifications or followed threads.

![kuva](https://user-images.githubusercontent.com/17256841/162534565-47301903-705d-43ee-949e-f2b1531b7b84.png)

Notifications section loading endlessly in the picture.

**Reply by mapx-** [reply]
you have to tell me "how to" (follow / get notifications etc)

**Reply by peace2000** [reply]
Just click that icon at the bottom-right, there's no need to really have anything to follow / have any notifications about anything.

**Reply by mapx-** [reply]
test

`ylilauta.org##+js(aost, document.createElement, /(?=(content|new))(?!.*(Dropdown|open).*)/, 2)`

**Reply by peace2000** [reply]
Toast gets triggered again on Firefox, works on Chrome.

**Reply by mapx-** [reply]
working fine on my end on FF too, do you still use `ScriptletDoctor` ?

**Reply by peace2000** [reply]
I don't

**Reply by mapx-** [reply]
all these `aost` filters are working in FF only using `Scriptlet Doctor`

https://github.com/JustOff/scriptlet-doctor/releases

**Reply by peace2000** [reply]
Ok, yeah got it to work with that addon. Too bad this solution can't be delivered for Finnish list users that use Firefox.

Maybe for Firefox users I could restyle that nag and make it transparent.

`ylilauta.org##.warning.toast:style(background-color: transparent !important; color: transparent !important)`
`ylilauta.org##.warning.toast h3:style(color: transparent !important)`

Is there a working way in CSS to make that nag window unclickable? `User-select: none` didn't work.

**Reply by peace2000** [reply]
Unfortunately noticed a new breakage concerning that `aost` filter. Cannot view any poll results: `https://ylilauta.org/poll/` click "show results" under any of those polls.

Edit, I also cannot create a poll.

**Reply by mapx-** [reply]
Test
`ylilauta.org##+js(aost, document.createElement, /(?=(content|new))(?!.*(Dropdown|open|Poll).*)/, 2)`

for FF just add the `noopframe` filter or recommend finnish users to install `ScriptDoctor` (adding that domain in SD options)

**Reply by peace2000** [reply]
Seems to work better, thanks! Didn't notice any breaking at least.

**Reply by peace2000** [reply]
They changed the site, doesn't work anymore. Causes errors.

**Reply by mapx-** [reply]
yeah, they are fighting back, use `noopframe` filter: your best choice

**Reply by peace2000** [reply]
Now even that went down. Any new ideas to this?

**Reply by mapx-** [reply]
```
||fcdn.lauta.media^$frame,redirect=noopframe,important,badfilter
*$script,domain=fcdn.lauta.media,redirect-rule=noopjs
```

**Reply by peace2000** [reply]
Thanks again :)
